### ISSUE TITLE
Typescript error when setting a ref

### ISSUE THREAD

**Issue by embpdaniel** [MrLoh/styled-native-components#19]
Steps to reproduce:

1. Create a styled component from a TouchableOpacity
2. Use the `useRef` hook and set it to null as default
3. Assign the ref to the styled component

Expected:
Typescript should not show errors on the ref prop

Actual:
Typescript displays the following error:

```bash
(JSX attribute) ref: React.MutableRefObject<null>
Type '{ children: Element; ref: MutableRefObject<null>; onPress: () => void; }' is not assignable to type 'IntrinsicAttributes & Omit<TouchableOpacityProps, never> & { children?: ReactNode; }'.
  Property 'ref' does not exist on type 'IntrinsicAttributes & Omit<TouchableOpacityProps, never> & { children?: ReactNode; }
```

Simplified reproduction:

```jsx
import React, { useRef } from 'react';
import styled from 'styled-native-components';

const Wrapper = styled.TouchableOpacity`
  margin-left: 2rem;
`;

const MyComponent = (): JSX.Element => {
  const ref = useRef(null);
  return (
    <Wrapper ref={ref} onPress={() => console.log('press')}>
      ...child components
    </Wrapper>
  );
};

export default MyComponent;
```
